code/modules/mob/new_player/new_player.dm

```diff
@@ -454,4 +463,6 @@
 		return FALSE
 	if(latejoin && !job.special_check_latejoin(client))
 		return FALSE
+	if(length(SSticker.mode.valid_job_types) && !(job.type in SSticker.mode.valid_job_types))
```

Comment:
Add this to the jobs subsystem too. No duplicate checks please. (currently it runs some twice)

Reply:
Can't add this in the job subsystem because the gamemode hasn't loaded yet and this would always fail.

Reply:
Yes, but it still logs here which it shouldnt (unless you removed that and I didn't see the commit)

Reply:
It doesn't log here as this is in the new_player proc, it just returns FALSE.